crates/ruff_linter/src/rules/fastapi/rules/fastapi_non_annotated_dependency.rs

```diff
@@ -153,44 +146,94 @@ fn is_fastapi_dependency(checker: &Checker, expr: &ast::Expr) -> bool {
+                "typing"
+            } else {
+                "typing_extensions"
+            };
+            let (import_edit, binding) = checker.importer().get_or_import_symbol(
+                &ImportRequest::import_from(module, "Annotated"),
+                parameter.range.start(),
+                checker.semantic(),
+            )?;
+
+            // Refine the match from `is_fastapi_dependency` to exclude Depends
+            // and Security, which don't have the same argument structure. The
+            // others need to be converted from `q: str = Query("")` to `q:
+            // Annotated[str, Query()] = ""` for example, but Depends and
+            // Security need to stay like `Annotated[str, Depends(callable)]`
+            let is_dep = checker
+                .semantic()
+                .resolve_qualified_name(map_callable(default))
+                .is_some_and(|qualified_name| {
+                    !matches!(
+                        qualified_name.segments(),
+                        ["fastapi", "Depends" | "Security"]
+                    )
+                });
+
+            // Each of these classes takes a single, optional default
+            // argument, followed by kw-only arguments
+            let default_arg = default
+                .as_call_expr()
+                .and_then(|args| args.arguments.find_argument("default", 0));
+
+            let kwarg_list: Option<Vec<_>> = default.as_call_expr().map(|args| {
+                args.arguments
+                    .keywords
+                    .iter()
+                    .filter_map(|kwarg| match kwarg.arg.as_ref() {
+                        None => None,
+                        Some(name) if name == "default" => None,
+                        Some(_) => Some(checker.locator().slice(kwarg.range())),
+                    })
+                    .collect()
+            });
+
+            let content = if is_dep && default_arg.is_some() {
+                let default_arg = default_arg.unwrap();
+                let kwarg_list = match kwarg_list {
+                    Some(v) => v.join(", "),
+                    None => String::new(),
                 };
-                let (import_edit, binding) = checker.importer().get_or_import_symbol(
-                    &ImportRequest::import_from(module, "Annotated"),
-                    parameter.range.start(),
-                    checker.semantic(),
-                )?;
-                let content = format!(
+                seen_default = true;
+                format!(
+                    "{}: {}[{}, {}({kwarg_list})] = {}",
+                    &parameter.parameter.name.id,
+                    binding,
+                    checker.locator().slice(annotation.range()),
+                    checker.locator().slice(map_callable(default).range()),
+                    checker.locator().slice(default_arg.value().range()),
+                )
```

Comment:
This is sort of hard to read. It would help readability to use named arguments
```suggestion
                    "{parameter_name}: {binding}[{annotation}, {default}({kwarg_list})] = {default_value}",
                    parameter_name = &parameter.parameter.name.id,
                    annotation = checker.locator().slice(annotation.range()),
                    metadata = checker.locator().slice(map_callable(default).range()),
                    default_valuechecker.locator().slice(default_arg.value().range()),
                )
```